File: inspector/src/main/groovy/com/jakeout/gradle/utils/StringUtil.java
package com.jakeout.gradle.utils;

import com.google.common.base.Function;
import com.google.common.base.Strings;

public class StringUtil {
    public static final String TAB_CHARACTER = "  ";

    public static Function<String, String> tabAppender(final int numTabs) {
        return new Function<String, String>() {
            public String apply(String s) {
                return Strings.repeat(TAB_CHARACTER, numTabs) + s.replace("\n", "\n" + Strings.repeat(TAB_CHARACTER, numTabs));
            }
        };
    }
}

File: inspector/src/main/kotlin/com/jakeout/gradle/utils/StringUtil.kt
package com.jakeout.gradle.utils

import com.google.common.base.Function
import com.google.common.base.Strings

public object StringUtil {

    val TAB_CHARACTER: String = "  "

    fun tabAppender(numTabs: Int): Function<String, String> {
        return object : Function<String, String> {
            override fun apply(s: String?): String? {
                return Strings.repeat(TAB_CHARACTER, numTabs) + s!!.replace("\n", "\n" + Strings.repeat(TAB_CHARACTER, numTabs))
            }
        }
    }
}
